sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/utils/SDKUtil.java

```diff
@@ -79,4 +98,75 @@ public static ArrayList listFiles(String sourceImageFolder,
     return (Object[]) input[i];
   }
 
+  public static List<CarbonFile> extractFilesFromFolder(String path,
+      String suf, Configuration hadoopConf) {
+    List dataFiles = listFiles(path, suf, hadoopConf);
+    List<CarbonFile> carbonFiles = new ArrayList<>();
+    for (Object dataFile: dataFiles) {
+      carbonFiles.add(FileFactory.getCarbonFile(dataFile.toString(), hadoopConf));
+    }
+    if (CollectionUtils.isEmpty(dataFiles)) {
+      throw new RuntimeException("No file found at given location. Please provide" +
+          "the correct folder location.");
+    }
+    return carbonFiles;
+  }
+
+  public static DataFileStream<GenericData.Record> buildAvroReader(CarbonFile carbonFile,
```

Comment:
If validate Files methods are refactored to respecitive type carbonwriter as given in another review comment, no longer require these type specific buildreaders in this file(SDKUtil.java). Same for other readers below(buildOrcReader, buildParquetReader, buildCsvParser, buildJsonReader)

Reply:
Done.